src/engine/plugins/viaroute.cpp

```diff
@@ -15,9 +15,12 @@
 namespace osrm::engine::plugins
 {
 
-ViaRoutePlugin::ViaRoutePlugin(int max_locations_viaroute, int max_alternatives)
+ViaRoutePlugin::ViaRoutePlugin(int max_locations_viaroute,
+                               int max_alternatives,
+                               const boost::optional<double> default_radius)
```

Comment:
If you will make it non-const you will be able to `std::move` it.

Reply:
Alright - I've taken a closer look at the passing of default_radius and applied copy & move across the board for default_radius.

Reply:
A clang-tidy run seems to say that applying std::move to boost::optional<double> is redundant. It's somewhat unexpected for me since although an -O flag in godbolt seems to resolve them the same way, it did yield different results without. I've gone ahead and removed the const qualifier here though nonetheless.